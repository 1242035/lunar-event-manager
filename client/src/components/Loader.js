import Spinner from 'react-loader-spinner';
import React from 'react';
import classNames from 'classnames';

/**
 * Component for showing a loading screen
 * @param{Object} props - passed properties
 * @return{ReactElement} - react element
 */
const Loader = (props) => {
  const classes = classNames({ 'io-modal': true, hide: !props.showLoader });

  return (
    <div className={classes}>
      <Spinner
        type="TailSpin"
        color="#424553"
        height="100"
        width="100"
      />
    </div>
  );
};

export default Loader;
