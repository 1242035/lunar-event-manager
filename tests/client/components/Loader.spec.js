import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../../../client/src/components/Loader';

describe('Loader', () => {
  let wrapper;

  it('Should maintain existing snapshot', () => {
    wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render component', () => {
    wrapper = shallow(<Loader />);
    expect(wrapper.length).toEqual(1);
  });
});
