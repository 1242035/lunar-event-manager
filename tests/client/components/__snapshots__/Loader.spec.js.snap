// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Should maintain existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loader
        color="#424553"
        height="100"
        type="TailSpin"
        visible={true}
        width="100"
      />,
      "className": "io-modal hide",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "#424553",
        "height": "100",
        "type": "TailSpin",
        "visible": true,
        "width": "100",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loader
          color="#424553"
          height="100"
          type="TailSpin"
          visible={true}
          width="100"
        />,
        "className": "io-modal hide",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#424553",
          "height": "100",
          "type": "TailSpin",
          "visible": true,
          "width": "100",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
